charms/istio-pilot/src/charm.py

```diff
@@ -140,7 +140,27 @@ def handle_default_gateway(self, event):
         """
         t = self.env.get_template('gateway.yaml.j2')
         gateway = self.model.config['default-gateway']
-        manifest = t.render(name=gateway, app_name=self.app.name)
+        secret_name = (
+            "istio-gw-secret"
```

Comment:
Can we name this dynamically based on the charm name, like `"{charm-name}-gateway-secret"`?  Feels a bit verbose, but less likely to collide with others (I could see someone writing the istio-gateway charm expecting they could own something called `istio-gw-secret`

Reply:
Agreed, preparing a new commit and will test now with these changes